modules/media_duplicate_validation/media_duplicate_validation.module

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @file
+ * Contains media_duplicate_validation.module.
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_help().
+ */
+function media_duplicate_validation_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the media_duplicate_validation module.
+    case 'help.page.media_duplicate_validation':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Media Validation plugins to help prevent duplication of media items') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_cache_flush().
+ */
+function media_duplicate_validation_cache_flush() {
+  /** @var \Drupal\media_duplicate_validation\Plugin\MediaDuplicateValidationManager $duplicate_manager */
+  $duplicate_manager = \Drupal::service('plugin.manager.media_duplicate_validation');
+  $duplicate_manager->buildPluginSchemas();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function media_duplicate_validation_media_insert(EntityInterface $entity) {
+  /** @var \Drupal\media_duplicate_validation\Plugin\MediaDuplicateValidationManager $duplicate_manager */
+  $duplicate_manager = \Drupal::service('plugin.manager.media_duplicate_validation');
+
+  foreach ($duplicate_manager->getDefinitions() as $definition) {
```

Comment:
This needs a code comment explaining what is a definition and why you loop through them.